pkg/status/status.go

```diff
@@ -159,6 +184,8 @@ func (s *status) setStatusCondition(condition configv1.ClusterStatusConditionTyp
 	case configv1.OperatorAvailable:
 		availableStatus = configv1.ConditionTrue
 		availableMessage = message
+		// Only update the version when the operator becomes available
+		s.setOperandVersion()
```

Comment:
A log message reflecting the current and previous version will be helpful. You can combine it with another existing log message if needed.

Reply:
Included in the latest commit.